### PR TITLE
[Network] received_characters rework

### LINKED ISSUE #2296: Re:start, Xkore 2 = wrong gender/sprite errors
Title says it all, when loading into a female character on restart with xkore 2 it shows as a male. I would like to fix this if a GM were to ask me to take a screenshot.

The other problem is sprite errors with xkore 2.  I think they are pulling something with invisible NPCs to crash bots.  When my bot goes into certain maps the client crashes saying a novice or thief sprite is missing and this is in biolabs where obviously neither of these things are around.  I confirmed with a normal client that there aren't people of these classes there when it occurs either.  

I think it's a way to screw with bots because when i was watching one of my bots in another area there was a weird NPC in the middle of the field that had a 99 aura around it and it's not normally there.  Anyone else encounter this?

### PR DESCRIPTION
**Pull request by alisonrag**
Rework in received_characters subs, now is more dinamic, use less code and will be easy to update

This will affect received_characters, charBlockSize, character_creation_successfull and xKore 2 map-server

- [x] Code Review
- [ ] Q.A
- [x] Test

Tested in bRO, iRO ReStart and kRO_RagexeRE_2017_06_14b
bRO:
bro_xkore2

iRO:
iro_xkore2

this fixes #2296

### CODE REVIEW

**Comment by AlexandreIV** [comment]
image

Tested in 2015_11_04a.

**Comment by lututui** [comment]
Currently openkore does not support job translation.
It's on my to-do list

**Comment by Nipodemos** [comment]
just tested and on bRO connected perfectly

File: src/Base/Ragnarok/CharServer.pm [hunk 1]
```diff
@@ -310,19 +155,20 @@ sub char_login {
...
 		$self->{sessionStore}->mark($session);
 		my @characters = $self->getCharacters();
-		if (!$characters[$args->{slot}]) {
+
+		if ($args->{slot} != $char->{slot}) {
 			# Invalid character selected.
 			$client->send(pack('C*', 0x6C, 0x00, 0));
 		} else {
-			my $char = $characters[$args->{slot}];
+			my $char = $char;
```
Comment (Nipodemos) [inline src/Base/Ragnarok/CharServer.pm]: i just wondering if need to define my $char again here...
does it needs?

**Comment by c4c1n6kr3m1** [comment]
1. For as long as I can remember , old char (for old account) on iRO will show sex as 99 , not 0 or 1,  that's why  $accountSex2 is used for old char

2. what if hand over character_creation_successful to received_characters , they are similar

**Comment by alisonrag** [comment]
updated like review, and i rechecked the code.

**Comment by c4c1n6kr3m1** [comment]
@alisonrag 
but you did not try with old IRO char created before this new sex system , right?
the reason i limit with /^0|1$/ is because old char will show 99, not 0 or 1
see for your self
```
<< Received packet:      099D [439 bytes]   Aug 26 12:47:21 2018
  0>  9D 09 B7 01 XX XX XX XX    DF 22 02 00 8C 82 15 00    ....XXXX."......
 16>  0B 7B 00 00 0C 00 00 00    00 00 00 00 00 00 00 00    .{..............
 32>  00 00 00 00 00 00 00 00    00 00 00 00 3F 00 5D 09    ............?.].
 48>  00 00 5D 09 00 00 F9 02    F9 02 96 00 08 00 0F 00    ..].............
 64>  00 00 48 00 02 00 00 00    00 00 68 03 00 00 07 00    ..H.......h.....
 80>  00 00 XX XX XX XX XX XX    XX XX XX XX XX XX XX XX    ..XXXXXXXXXXXXXX
 96>  XX XX XX XX XX XX XX XX    XX XX 01 50 01 1E 2D 01    XXXXXXXXXX.P..-.
112>  00 00 00 00 68 75 67 65    6C 2E 67 61 74 00 00 00    ....hugel.gat...
128>  00 00 00 00 00 00 00 00    00 00 00 00 00 00 00 00    ................
144>  00 00 00 00 63 XX XX XX    XX 6B FC 22 00 82 9B 0F    ....cXXXXk."....
160>  00 01 00 B4 00 32 00 00    00 00 00 00 00 00 00 00    .....2..........
176>  00 00 00 00 00 00 00 00    00 00 00 00 00 97 00 ED    ................
192>  18 00 00 ED 18 00 00 A3    05 A3 05 96 00 08 00 03    ................
208>  00 00 00 63 00 17 00 00    00 00 00 AA 02 00 00 06    ...c............
224>  00 00 00 XX XX XX XX XX    XX XX XX XX XX XX XX XX    ...XXXXXXXXXXXXX
240>  XX XX XX XX XX XX XX XX    XX XX XX 01 50 32 50 19    XXXXXXXXXXX.P2P.
256>  01 01 00 00 00 70 72 6F    6E 74 65 72 61 2E 67 61    .....prontera.ga
272>  74 00 00 00 00 00 00 00    00 00 00 00 00 00 00 00    t...............
288>  00 00 00 00 00 63 XX XX    XX XX 3C 0A 00 00 31 1E    .....cXXXX<...1.
304>  09 00 34 00 00 00 0A 00    00 00 00 00 00 00 00 00    ..4.............
320>  00 00 00 00 00 00 00 00    00 00 00 00 00 00 04 00    ................
336>  62 00 00 00 62 00 00 00    25 00 25 00 96 00 05 00    b...b...%.%.....
352>  10 00 00 00 09 00 00 00    00 00 00 00 68 03 00 00    ............h...
368>  08 00 00 00 XX XX XX XX    XX XX XX XX XX XX XX XX    ....XXXXXXXXXXXX
384>  XX XX XX XX XX XX XX XX    XX XX XX XX 01 04 01 01    XXXXXXXXXXXX....
400>  19 01 02 00 00 00 69 7A    6C 75 64 65 2E 67 61 74    ......izlude.gat
416>  00 00 00 00 00 00 00 00    00 00 00 00 00 00 00 00    ................
432>  00 00 00 00 00 00 63                                  ......c
```

**Comment by alisonrag** [comment]
So if 99 we have to use accountID?

**Comment by c4c1n6kr3m1** [comment]
at that time,  my conclusion is, if not 0 or 1 , use accountID

**Comment by Fadreus** [comment]
I test in iRO Chaos renewal, stuck on character selection (Timeout).

**Pull request merged**